CancelWorkflowInstanceConcurrentlyTest.shouldCancelAfterJobComplete unstable

**Description**

```
[ERROR] Tests run: 16, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 142.486 s <<< FAILURE! - in io.zeebe.engine.processor.workflow.instance.CancelWorkflowInstanceConcurrentlyTest

[ERROR] io.zeebe.engine.processor.workflow.instance.CancelWorkflowInstanceConcurrentlyTest.shouldCancelAfterJobComplete[multi-instance]  Time elapsed: 17.138 s  <<< ERROR!

io.zeebe.test.util.stream.StreamWrapperException: No event found matching the criteria

	at java.base/java.util.Optional.orElseThrow(Optional.java:408)

	at io.zeebe.test.util.stream.StreamWrapper.getFirst(StreamWrapper.java:77)

	at io.zeebe.engine.util.client.WorkflowInstanceClient$WorkflowInstanceCreationClient.create(WorkflowInstanceClient.java:83)

	at io.zeebe.engine.processor.workflow.instance.CancelWorkflowInstanceConcurrentlyTest.init(CancelWorkflowInstanceConcurrentlyTest.java:132)

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

	at java.base/java.lang.reflect.Method.invoke(Method.java:566)

	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)

	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)

	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)

	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)

	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)

	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)

	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)

	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)

	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)

	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)

	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)

	at org.junit.runners.Suite.runChild(Suite.java:128)

	at org.junit.runners.Suite.runChild(Suite.java:27)

	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)

	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)

	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)

	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)

	at org.junit.rules.RunRules.evaluate(RunRules.java:20)

	at org.junit.rules.RunRules.evaluate(RunRules.java:20)

	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)

	at org.junit.runners.Suite.runChild(Suite.java:128)

	at org.junit.runners.Suite.runChild(Suite.java:27)

	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)

	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)

	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)

	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)

	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)

	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)

	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)

	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)

	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
```

Stacktraces and standard output:

<details>
<summary>shouldCancelAfterJobComplete()</summary>
<p>

```log
io.zeebe.test.util.stream.StreamWrapperException: No event found matching the criteria
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at io.zeebe.test.util.stream.StreamWrapper.getFirst(StreamWrapper.java:77)
	at io.zeebe.engine.util.client.DeploymentClient.lambda$static$1(DeploymentClient.java:37)
	at io.zeebe.engine.util.client.DeploymentClient.deploy(DeploymentClient.java:139)
	at io.zeebe.engine.processing.workflowinstance.CancelWorkflowInstanceConcurrentlyTest.init(CancelWorkflowInstanceConcurrentlyTest.java:127)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```

```log
07:49:25.427 [] INFO  io.zeebe.test.records - Test failed, following records were exported:
```
</p>
</details>  

<details>
<summary>shouldCancelAfterJobCompleted()</summary>
<p>

```log
io.zeebe.test.util.stream.StreamWrapperException: No event found matching the criteria
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at io.zeebe.test.util.stream.StreamWrapper.getFirst(StreamWrapper.java:77)
	at io.zeebe.engine.util.client.DeploymentClient.lambda$static$1(DeploymentClient.java:37)
	at io.zeebe.engine.util.client.DeploymentClient.deploy(DeploymentClient.java:139)
	at io.zeebe.engine.processing.workflowinstance.CancelWorkflowInstanceConcurrentlyTest.init(CancelWorkflowInstanceConcurrentlyTest.java:127)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```

```log
07:49:15.418 [] INFO  io.zeebe.test.records - Test failed, following records were exported:
```
</p>
</details>  

<details>
<summary>shouldCancelAfterElementCompleting() (only one with real output logs)</summary>
<p>

```log
java.lang.AssertionError: 

Expecting:
 <[]>
to contain subsequence:
 <["task", "task", "process"]>

	at io.zeebe.engine.processing.workflowinstance.CancelWorkflowInstanceConcurrentlyTest.shouldCancelAfterElementCompleting(CancelWorkflowInstanceConcurrentlyTest.java:236)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```

```log
07:49:04.703 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@3c96fdc4)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@6cb90c22, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@2ca6b2d9, configuration: Configuration(false)]
07:49:04.727 [] DEBUG io.zeebe.util.buffer - Close stream processor
07:49:04.728 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-1.
07:49:04.766 [] INFO  io.zeebe.logstreams - Closed stream stream-1
07:49:05.342 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.logstreams - Recovering state of partition 1 from snapshot
07:49:05.343 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@11b29994)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@3c3d2d35, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@1dcb1f31, configuration: Configuration(false)]
07:49:05.344 [Broker-0-StreamProcessor-1] INFO  io.zeebe.logstreams - Recovered state of partition 1 from snapshot at position -1
07:49:05.345 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@7a6fba69)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@7f94e289, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@2a153ccb, configuration: Configuration(false)]
07:49:05.345 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@6f628c2f)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@6859df58, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@3001f870, configuration: Configuration(false)]
07:49:05.347 [Broker-0-StreamProcessor-1] INFO  io.zeebe.processor - Processor starts reprocessing, until last source event position 505
07:49:05.355 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@6e645b36)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@58318638, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@3fc66d0a, configuration: Configuration(false)]
07:49:05.368 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=ELEMENT_COMPLETED, elementId=task, bpmnElementType=SERVICE_TASK, elementInstanceKey=2251799813685258, flowScopeKey=2251799813685251, workflowInstanceKey=2251799813685251, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=1, workflowKey=2251799813685249}]
07:49:05.377 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@75b9aebc)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@2b7ef73e, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@4471c806, configuration: Configuration(false)]
07:49:05.387 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=ELEMENT_COMPLETING, elementId=task, bpmnElementType=SERVICE_TASK, elementInstanceKey=2251799813685268, flowScopeKey=2251799813685261, workflowInstanceKey=2251799813685261, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=1, workflowKey=2251799813685249}]
07:49:05.397 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@3a4264d5)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@7a93452c, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@1520fa10, configuration: Configuration(false)]
07:49:05.408 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=SEQUENCE_FLOW_TAKEN, elementId=sequenceFlow_1266a4db-c5c0-4a02-ac1e-1c4c4159ab5f, bpmnElementType=SEQUENCE_FLOW, elementInstanceKey=2251799813685280, flowScopeKey=2251799813685271, workflowInstanceKey=2251799813685271, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=1, workflowKey=2251799813685249}]
07:49:05.416 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@d490036)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@67376651, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@7793a009, configuration: Configuration(false)]
07:49:05.433 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@4af392af)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@19402f5b, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@6ece3a3b, configuration: Configuration(false)]
07:49:05.452 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=ELEMENT_COMPLETED, elementId=task, bpmnElementType=SERVICE_TASK, elementInstanceKey=2251799813685302, flowScopeKey=2251799813685293, workflowInstanceKey=2251799813685293, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=2, workflowKey=2251799813685291}]
07:49:05.462 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@7ae3bb47)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@5115423f, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@3bb2c6ce, configuration: Configuration(false)]
07:49:05.472 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=ELEMENT_COMPLETING, elementId=task, bpmnElementType=SERVICE_TASK, elementInstanceKey=2251799813685315, flowScopeKey=2251799813685306, workflowInstanceKey=2251799813685306, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=2, workflowKey=2251799813685291}]
07:49:05.483 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@2ff3a974)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@32ec2185, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@8e54bd0, configuration: Configuration(false)]
07:49:05.493 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=SEQUENCE_FLOW_TAKEN, elementId=sequenceFlow_2f887543-c7e7-4d88-b523-34507343a3d3, bpmnElementType=SEQUENCE_FLOW, elementInstanceKey=2251799813685331, flowScopeKey=2251799813685319, workflowInstanceKey=2251799813685319, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=2, workflowKey=2251799813685291}]
07:49:05.503 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@4e62fb7c)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@748e24e6, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@6fd5eb39, configuration: Configuration(false)]
07:49:05.522 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@3fa0bcf3)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@4165e168, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@65f148d6, configuration: Configuration(false)]
07:49:05.542 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=SEQUENCE_FLOW_TAKEN, elementId=sequenceFlow_06e9aeb8-88fb-400b-8eae-abfc93ca000e, bpmnElementType=SEQUENCE_FLOW, elementInstanceKey=2251799813685362, flowScopeKey=2251799813685352, workflowInstanceKey=2251799813685347, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=3, workflowKey=2251799813685345}]
07:49:05.552 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@455a53ad)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@17846133, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@744175d9, configuration: Configuration(false)]
07:49:05.565 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=ELEMENT_COMPLETED, elementId=task, bpmnElementType=SERVICE_TASK, elementInstanceKey=2251799813685376, flowScopeKey=2251799813685369, workflowInstanceKey=2251799813685364, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=3, workflowKey=2251799813685345}]
07:49:05.576 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@c988d58)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@59d7714d, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@70922cb5, configuration: Configuration(false)]
07:49:05.619 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=ELEMENT_ACTIVATING, elementId=endEvent_b0f6989e-f04b-4d19-a0d1-2cc884ac78e5, bpmnElementType=END_EVENT, elementInstanceKey=2251799813685396, flowScopeKey=2251799813685385, workflowInstanceKey=2251799813685380, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=3, workflowKey=2251799813685345}]
07:49:05.634 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@7a37c8d0)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@3cb0f658, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@5a783e83, configuration: Configuration(false)]
07:49:05.648 [Broker-0-StreamProcessor-1] DEBUG io.zeebe.broker.workflow - Don't process event because of an illegal state transition: Expected flow scope instance to be in state 'ELEMENT_ACTIVATED' but was 'ELEMENT_TERMINATING'. [context: {intent=ELEMENT_COMPLETING, elementId=task, bpmnElementType=SERVICE_TASK, elementInstanceKey=2251799813685410, flowScopeKey=2251799813685403, workflowInstanceKey=2251799813685398, parentWorkflowInstanceKey=-1, parentElementInstanceKey=-1, bpmnProcessId=process, workflowVersion=3, workflowKey=2251799813685345}]
07:49:05.664 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@243e5118)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@324331b7, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@70d2ca15, configuration: Configuration(false)]
07:49:05.814 [Broker-0-StreamProcessor-1] ERROR io.zeebe.logstreams - Unexpected error on recovery happens.
io.zeebe.engine.processing.streamprocessor.InconsistentReprocessingException: Reprocessing issue detected! Restore the data from a backup and follow the recommended upgrade procedure. [cause: "Records were created on reprocessing but not written on the log stream.", log-stream-record: {"partitionId":1,"value":{"type":"test","errorMessage":"","deadline":-1,"variables":{},"errorCode":"","retries":3,"elementInstanceKey":2251799813685421,"customHeaders":{},"worker":"","workflowDefinitionVersion":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413,"elementId":"task"},"key":2251799813685425,"sourceRecordPosition":504,"recordType":"EVENT","rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","valueType":"JOB","intent":"COMPLETED","timestamp":1601884144767,"position":505}, reprocessing-record: {key=2251799813685426, sourceRecordPosition=502, intent=JobIntent:CREATED, recordType=EVENT}]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.lambda$verifyRecordMatchesToReprocessing$9(ReProcessingStateMachine.java:388) ~[classes/:?]
	at java.util.Optional.ifPresent(Optional.java:183) ~[?:?]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.verifyRecordMatchesToReprocessing(ReProcessingStateMachine.java:386) ~[classes/:?]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.reprocessEvent(ReProcessingStateMachine.java:269) ~[classes/:?]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.reprocessNextEvent(ReProcessingStateMachine.java:237) ~[classes/:?]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:73) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:118) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:204) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
07:49:05.831 [Broker-0-StreamProcessor-1] ERROR io.zeebe.logstreams - Actor Broker-0-StreamProcessor-1 failed in phase STARTED.
io.zeebe.engine.processing.streamprocessor.InconsistentReprocessingException: Reprocessing issue detected! Restore the data from a backup and follow the recommended upgrade procedure. [cause: "Records were created on reprocessing but not written on the log stream.", log-stream-record: {"partitionId":1,"value":{"type":"test","errorMessage":"","deadline":-1,"variables":{},"errorCode":"","retries":3,"elementInstanceKey":2251799813685421,"customHeaders":{},"worker":"","workflowDefinitionVersion":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413,"elementId":"task"},"key":2251799813685425,"sourceRecordPosition":504,"recordType":"EVENT","rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","valueType":"JOB","intent":"COMPLETED","timestamp":1601884144767,"position":505}, reprocessing-record: {key=2251799813685426, sourceRecordPosition=502, intent=JobIntent:CREATED, recordType=EVENT}]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.lambda$verifyRecordMatchesToReprocessing$9(ReProcessingStateMachine.java:388) ~[classes/:?]
	at java.util.Optional.ifPresent(Optional.java:183) ~[?:?]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.verifyRecordMatchesToReprocessing(ReProcessingStateMachine.java:386) ~[classes/:?]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.reprocessEvent(ReProcessingStateMachine.java:269) ~[classes/:?]
	at io.zeebe.engine.processing.streamprocessor.ReProcessingStateMachine.reprocessNextEvent(ReProcessingStateMachine.java:237) ~[classes/:?]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:73) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:118) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:204) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
07:49:10.356 [] INFO  io.zeebe.test.records - Test failed, following records were exported:
07:49:10.375 [] INFO  io.zeebe.test.records - {"valueType":"DEPLOYMENT","key":-1,"position":479,"timestamp":1601884144695,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfODA4MmVjNzUtYjEwMC00ZjFmLTk2ZjUtNTZkZTQxZDlhYTdlIiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvY2VzcyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdGFydEV2ZW50IGlkPSJzdGFydEV2ZW50X2JhZGIyN2VjLWM5NGMtNDUzNi1iOWEwLWQ2OGFmZjVlZDczZSI+CiAgICAgIDxvdXRnb2luZz5zZXF1ZW5jZUZsb3dfNmY...
07:49:10.378 [] INFO  io.zeebe.test.records - {"valueType":"DEPLOYMENT","key":2251799813685414,"position":480,"timestamp":1601884144714,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfODA4MmVjNzUtYjEwMC00ZjFmLTk2ZjUtNTZkZTQxZDlhYTdlIiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvY2VzcyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdGFydEV2ZW50IGlkPSJzdGFydEV2ZW50X2JhZGIyN2VjLWM5NGMtNDUzNi1iOWEwLWQ2OGFmZjVlZDczZSI+CiAgICAgIDxvdXRnb2luZz5zZXF1ZW...
07:49:10.379 [] INFO  io.zeebe.test.records - {"valueType":"DEPLOYMENT","key":2251799813685414,"position":481,"timestamp":1601884144715,"recordType":"COMMAND","intent":"DISTRIBUTE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfODA4MmVjNzUtYjEwMC00ZjFmLTk2ZjUtNTZkZTQxZDlhYTdlIiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvY2VzcyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdGFydEV2ZW50IGlkPSJzdGFydEV2ZW50X2JhZGIyN2VjLWM5NGMtNDUzNi1iOWEwLWQ2OGFmZjVlZDczZSI+CiAgICAgIDxvdXRnb2luZz5zZ...
07:49:10.380 [] INFO  io.zeebe.test.records - {"valueType":"DEPLOYMENT","key":2251799813685414,"position":482,"timestamp":1601884144716,"recordType":"EVENT","intent":"DISTRIBUTED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfODA4MmVjNzUtYjEwMC00ZjFmLTk2ZjUtNTZkZTQxZDlhYTdlIiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvY2VzcyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdGFydEV2ZW50IGlkPSJzdGFydEV2ZW50X2JhZGIyN2VjLWM5NGMtNDUzNi1iOWEwLWQ2OGFmZjVlZDczZSI+CiAgICAgIDxvdXRnb2luZz5zZX...
07:49:10.384 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE_CREATION","key":-1,"position":483,"timestamp":1601884144717,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":-1,"variables":{"items":["one","two"]},"bpmnProcessId":"process","workflowInstanceKey":-1,"workflowKey":-1},"sourceRecordPosition":-1}
07:49:10.388 [] INFO  io.zeebe.test.records - {"valueType":"VARIABLE","key":2251799813685416,"position":484,"timestamp":1601884144718,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"name":"items","value":"[\"one\",\"two\"]","scopeKey":2251799813685415,"workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413},"sourceRecordPosition":483}
07:49:10.391 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685415,"position":485,"timestamp":1601884144718,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":-1,"workflowKey":2251799813685413,"bpmnElementType":"PROCESS","elementId":"process","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":483}
07:49:10.397 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE_CREATION","key":2251799813685417,"position":486,"timestamp":1601884144718,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"variables":{"items":["one","two"]},"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413},"sourceRecordPosition":483}
07:49:10.398 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685415,"position":487,"timestamp":1601884144718,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":-1,"workflowKey":2251799813685413,"bpmnElementType":"PROCESS","elementId":"process","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":485}
07:49:10.398 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685418,"position":488,"timestamp":1601884144719,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685415,"workflowKey":2251799813685413,"bpmnElementType":"START_EVENT","elementId":"startEvent_badb27ec-c94c-4536-b9a0-d68aff5ed73e","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":487}
07:49:10.398 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685418,"position":489,"timestamp":1601884144719,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685415,"workflowKey":2251799813685413,"bpmnElementType":"START_EVENT","elementId":"startEvent_badb27ec-c94c-4536-b9a0-d68aff5ed73e","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":488}
07:49:10.399 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685418,"position":490,"timestamp":1601884144720,"recordType":"EVENT","intent":"ELEMENT_COMPLETING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685415,"workflowKey":2251799813685413,"bpmnElementType":"START_EVENT","elementId":"startEvent_badb27ec-c94c-4536-b9a0-d68aff5ed73e","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":489}
07:49:10.399 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685418,"position":491,"timestamp":1601884144720,"recordType":"EVENT","intent":"ELEMENT_COMPLETED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685415,"workflowKey":2251799813685413,"bpmnElementType":"START_EVENT","elementId":"startEvent_badb27ec-c94c-4536-b9a0-d68aff5ed73e","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":490}
07:49:10.399 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685419,"position":492,"timestamp":1601884144721,"recordType":"EVENT","intent":"SEQUENCE_FLOW_TAKEN","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685415,"workflowKey":2251799813685413,"bpmnElementType":"SEQUENCE_FLOW","elementId":"sequenceFlow_6f048459-211a-4f22-ad3c-917a2d4be5da","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":491}
07:49:10.399 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685420,"position":493,"timestamp":1601884144721,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685415,"workflowKey":2251799813685413,"bpmnElementType":"MULTI_INSTANCE_BODY","elementId":"task","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":492}
07:49:10.402 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685420,"position":494,"timestamp":1601884144722,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685415,"workflowKey":2251799813685413,"bpmnElementType":"MULTI_INSTANCE_BODY","elementId":"task","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":493}
07:49:10.403 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685421,"position":495,"timestamp":1601884144724,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685420,"workflowKey":2251799813685413,"bpmnElementType":"SERVICE_TASK","elementId":"task","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":494}
07:49:10.403 [] INFO  io.zeebe.test.records - {"valueType":"VARIABLE","key":2251799813685422,"position":496,"timestamp":1601884144724,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"name":"loopCounter","value":"1","scopeKey":2251799813685421,"workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413},"sourceRecordPosition":494}
07:49:10.403 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685423,"position":497,"timestamp":1601884144724,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685420,"workflowKey":2251799813685413,"bpmnElementType":"SERVICE_TASK","elementId":"task","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":494}
07:49:10.404 [] INFO  io.zeebe.test.records - {"valueType":"VARIABLE","key":2251799813685424,"position":498,"timestamp":1601884144724,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"name":"loopCounter","value":"2","scopeKey":2251799813685423,"workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413},"sourceRecordPosition":494}
07:49:10.404 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685421,"position":499,"timestamp":1601884144725,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685420,"workflowKey":2251799813685413,"bpmnElementType":"SERVICE_TASK","elementId":"task","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":495}
07:49:10.404 [] INFO  io.zeebe.test.records - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685423,"position":500,"timestamp":1601884144725,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"version":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"flowScopeKey":2251799813685420,"workflowKey":2251799813685413,"bpmnElementType":"SERVICE_TASK","elementId":"task","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":497}
07:49:10.405 [] INFO  io.zeebe.test.records - {"valueType":"JOB","key":-1,"position":501,"timestamp":1601884144726,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"type":"test","errorMessage":"","deadline":-1,"variables":{},"errorCode":"","retries":3,"elementInstanceKey":2251799813685421,"customHeaders":{},"worker":"","workflowDefinitionVersion":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413,"elementId":"task"},"sourceRecordPosition":499}
07:49:10.405 [] INFO  io.zeebe.test.records - {"valueType":"JOB","key":-1,"position":502,"timestamp":1601884144726,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"type":"test","errorMessage":"","deadline":-1,"variables":{},"errorCode":"","retries":3,"elementInstanceKey":2251799813685423,"customHeaders":{},"worker":"","workflowDefinitionVersion":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413,"elementId":"task"},"sourceRecordPosition":500}
07:49:10.405 [] INFO  io.zeebe.test.records - {"valueType":"JOB","key":2251799813685425,"position":503,"timestamp":1601884144726,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"0.25.0","value":{"type":"test","errorMessage":"","deadline":-1,"variables":{},"errorCode":"","retries":3,"elementInstanceKey":2251799813685421,"customHeaders":{},"worker":"","workflowDefinitionVersion":4,"bpmnProcessId":"process","workflowInstanceKey":2251799813685415,"workflowKey":2251799813685413,"elementId":"task"},"sourceRecordPosition":501}
```
</p>
</details>  

<details>
<summary>shouldCancelAfterElementCompleted()</summary>
<p>

```log
io.zeebe.test.util.stream.StreamWrapperException: No event found matching the criteria
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at io.zeebe.test.util.stream.StreamWrapper.getFirst(StreamWrapper.java:77)
	at io.zeebe.engine.util.client.DeploymentClient.lambda$static$1(DeploymentClient.java:37)
	at io.zeebe.engine.util.client.DeploymentClient.deploy(DeploymentClient.java:139)
	at io.zeebe.engine.processing.workflowinstance.CancelWorkflowInstanceConcurrentlyTest.init(CancelWorkflowInstanceConcurrentlyTest.java:127)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```

```log
07:49:20.423 [] INFO  io.zeebe.test.records - Test failed, following records were exported:
```
</p>
</details>

--- a/.gitignore
+++ b/.gitignore
@@ -26,6 +26,7 @@ faces-config.NavData
 *.DS_Store
 .java-version
 .gradle
+.worktree
 work/
 public/
 tngp-data

--- a/engine/src/test/java/io/zeebe/engine/processing/workflowinstance/CancelWorkflowInstanceConcurrentlyTest.java
+++ b/engine/src/test/java/io/zeebe/engine/processing/workflowinstance/CancelWorkflowInstanceConcurrentlyTest.java
@@ -144,6 +144,7 @@ public final class CancelWorkflowInstanceConcurrentlyTest {
     createdJob =
         RecordingExporter.jobRecords(JobIntent.CREATED)
             .withWorkflowInstanceKey(workflowInstanceKey)
+            .limit(2)
             .withType(JOB_TYPE)
             .getFirst();
 

